src/main/java/org/folio/repository/kbcredentials/KbCredentialsRepositoryImpl.java

```diff
@@ -56,6 +64,29 @@
     return mapResult(promise.future().recover(excTranslator.translateOrPassBy()), this::mapCredentialsCollection);
   }
 
+  @Override
+  public CompletableFuture<DbKbCredentials> save(DbKbCredentials credentials, String tenant) {
+    String query = format(INSERT_CREDENTIALS_QUERY, getKbCredentialsTableName(tenant));
+
+    String id = UUID.randomUUID().toString();
```

Comment:
![image](https://user-images.githubusercontent.com/42244720/78375032-e4c73b00-75d4-11ea-8aa5-289ba1f12dea.png)
It's a common pattern on Folio to allow id to be passed outside (in the request) for a new record.
If it's not present then generate it

Reply:
Fixed